packages/preload/src/index.ts

```diff
@@ -245,9 +252,22 @@ function initExposure(): void {
   contextBridge.exposeInMainWorld('restartPod', async (engine: string, podId: string): Promise<void> => {
     return ipcInvoke('container-provider-registry:restartPod', engine, podId);
   });
-  contextBridge.exposeInMainWorld('generatePodmanKube', async (engine: string, names: string[]): Promise<string> => {
-    return ipcInvoke('container-provider-registry:generatePodmanKube', engine, names);
-  });
+  contextBridge.exposeInMainWorld(
+    'generatePodmanKube',
+    async (engine: string, names: string[], kubeGeneratorId?: string): Promise<string> => {
+      return ipcInvoke('container-provider-registry:generatePodmanKube', engine, names, kubeGeneratorId);
+    },
+  );
+
+  contextBridge.exposeInMainWorld(
+    'generateKube',
+    async (
+      kubeGeneratorId: string,
+      kubernetesGeneratorArgument: KubernetesGeneratorArgument,
```

Comment:
shouldn't we have kubeGeneratorId as last parameter and being optional ?
```suggestion
      kubernetesGeneratorArgument: KubernetesGeneratorArgument,
      kubeGeneratorId?: string,
```

Reply:
Depends, do we consider Podman generator to be the default one ? and being called when kubeGeneratorId is undefined ?

Reply:
I think that there is room for a default one, maybe could be configured later also but yes generally there should always be a default (if matching the expectation and if available)

Reply:
How do we define the default, can an extension ask to be the default one ?
In the current implementation, simply called manually the podman `containerProviderRegistry.generatePodmanKube` ? We lost the information about the engine, so we should add it to the kubernetesGeneratorArgument ?

Reply:
so yes I think that if we introduce a property to name the default, we would have the name of the default generator persisted.

then, an extension can read that property as well.

yes it looks like engineId is important so it looks it requires another changes, when you specify the list of pods, compose, and containers, you need to specify both engineId and then the list of containers for this engineId, etc.

```ts
[
   {
      "engineId":"abcd",
      "containers":[
         "1",
         "2",
         "3"
      ],
      "pods":[
         "1"
      ],
      "compose":[
         "2"
      ]
   },
   {
      "engineId":"fgh",
      "pods":[
         "1"
      ]
   }
]
```